src/MoBi.Presentation/Tasks/MoBiFormulaTask.cs

```diff
@@ -129,32 +163,55 @@ public IMoBiCommand ChangeVariableName(SumFormula formula, string newVariableNam
 
       public IMoBiCommand AddValuePoint(TableFormula formula, ValuePoint newValuePoint, IBuildingBlock buildingBlock)
       {
-         return new AddValuePointCommand(formula, newValuePoint, buildingBlock).Run(_context);
+         var command = new AddValuePointCommand(formula, newValuePoint, buildingBlock).Run(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command, formula, buildingBlock);
       }
 
       public IMoBiCommand EditUseDerivedValues(TableFormula formula, bool newValue, bool oldValue, IBuildingBlock buildingBlock)
       {
-         return new EditUseDerivedValuesCommand(formula, newValue, oldValue, buildingBlock).Run(_context);
+         var command = new EditUseDerivedValuesCommand(formula, newValue, oldValue, buildingBlock).Run(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command, formula, buildingBlock);
       }
 
       public IMoBiCommand RemoveValuePointFromTableFormula(TableFormula formula, ValuePoint valuePoint, IBuildingBlock buildingBlock)
       {
-         return new RemoveValuePointFromTableFormulaCommand(formula, valuePoint, buildingBlock).Run(_context);
+         var command = new RemoveValuePointFromTableFormulaCommand(formula, valuePoint, buildingBlock).Run(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command, formula, buildingBlock);
       }
 
       public IMoBiCommand ChangeOffsetObject(TableFormulaWithOffset formula, IFormulaUsablePath path, IBuildingBlock buildingBlock)
       {
-         return new ChangeTableFormulaWithOffsetOffsetObjectPathCommand(formula, path, buildingBlock).Run(_context);
+         var command = new ChangeTableFormulaWithOffsetOffsetObjectPathCommand(formula, path, buildingBlock).Run(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command, formula, buildingBlock);
       }
 
       public IMoBiCommand ChangeTableObject(TableFormulaWithOffset formula, IFormulaUsablePath path, IBuildingBlock buildingBlock)
       {
-         return new ChangeTableFormulaWithOffsetTableObjectPathCommand(formula, path, buildingBlock).Run(_context);
+         var command = new ChangeTableFormulaWithOffsetTableObjectPathCommand(formula, path, buildingBlock).Run(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command, formula, buildingBlock);
       }
 
       public IMoBiCommand SetConstantFormulaValue(ConstantFormula formula, double kernelValue, Unit newDisplayUnit, Unit oldDisplayUnit, IBuildingBlock buildingBlock, IEntity formulaOwner)
       {
-         return new SetConstantFormulaValueCommand(formula, kernelValue, newDisplayUnit, oldDisplayUnit, buildingBlock, formulaOwner).Run(_context);
+         var command = new SetConstantFormulaValueCommand(formula, kernelValue, newDisplayUnit, oldDisplayUnit, buildingBlock, formulaOwner).Run(_context);
+         return withUpdatedDefaultStateAndValueOrigin(command, formula, buildingBlock);
```

Comment:
I think those are all the methods where update of state and value origin for depending parmaeters are required.... to check